discussion_agents/cog/functional/react.py

```diff
@@ -42,20 +43,22 @@ def _prompt_agent(llm: BaseChatModel, question: str, scratchpad: str) -> str:
         llm (BaseChatModel): The language model to be prompted.
         question (str): The question to ask the language model.
         scratchpad (str): Additional context or information for the language model.
-
+        examples (str): The example used for specific benchmark.
+        bench_type (str): Type of benchmark used for specific cases.
     Returns:
         str: The processed response from the language model.
     """
-    prompt = _build_agent_prompt(question=question, scratchpad=scratchpad)
+    prompt = _build_agent_prompt(question=question, scratchpad=scratchpad, examples=examples, instruction=instruction)
     out = llm(
         [
             HumanMessage(
                 content=prompt,
             )
-        ]
+        ],
+        stop = '\n'
     ).content
     assert isinstance(out, str)
-    return remove_newline(out)
+    return out 
```

Comment:
how come remove_newline is not used?

Reply:
because i have a stop = \n in the prompt_agent
so it stops when meeting \n therefore not necessary for remove_newline